tests/validators/test_enums.py

```diff
@@ -267,3 +267,21 @@ class MyEnum(Enum):
 
     with pytest.raises(SchemaError, match='`members` should have length > 0'):
         SchemaValidator(core_schema.enum_schema(MyEnum, []))
+
+
+def test_enum_with_str_subclass():
+    class MyEnum(Enum):
+        a = 'a'
+        b = 'b'
+
+    v = SchemaValidator(core_schema.enum_schema(MyEnum, list(MyEnum.__members__.values()), sub_type='str'))
```

Comment:
Test passes on `main` for me unless you remove the sub-type.

```suggestion
    v = SchemaValidator(core_schema.enum_schema(MyEnum, list(MyEnum.__members__.values()))
```